consult-org.el

```diff
@@ -0,0 +1,167 @@
+                          (number-sequence ?1 ?9))
+                  (mapcar (lambda (c) (cons c (format "Priority %c" c)))
+                          (number-sequence (max ?A org-highest-priority)
+                                           (min ?Z org-lowest-priority)))
+                  todo-keywords))))
+
+(defun consult-org--entries (match scope &rest skip)
+  "Return a list of consult locations from Org entries.
+
+MATCH, SCOPE and SKIP are as in `org-map-entries'."
+  (let (opoint line buffer)
+    (apply
+     #'org-map-entries
+     (lambda ()
+       (unless (eq buffer (current-buffer))
+         (setq opoint (point-min))
+         (setq line 1)
+         (setq buffer (current-buffer))
+         (setq org-outline-path-cache nil))
+       (setq line (+ line (consult--count-lines (prog1 (point)
+                                                  (goto-char opoint)))))
+       (setq opoint (point))
+       (pcase-let ((`(_ ,level ,todo ,prio) (org-heading-components)))
+         (consult--location-candidate
+          (org-format-outline-path (org-get-outline-path t t))
+          (point-marker)
+          line
+          'consult-org--level level
+          'consult-org--todo todo
+          'consult-org--priority prio)))
+     match scope skip)))
+
+;;;###autoload
+(defun consult-org-heading (&optional match scope)
+  "Jump to an Org heading.
+
+MATCH and SCOPE are as in `org-map-entries' and determine which
+entries are offered.  By default, all entries of the current
+buffer are offered."
+  (interactive (unless (derived-mode-p 'org-mode)
+                 (user-error "Must be called from an Org buffer.")))
+  (consult--jump
+   (consult--read
+    (consult--with-increased-gc (consult-org--entries match scope))
+    :prompt "Go to heading: "
+    :category 'consult-location
+    :sort nil
+    :title (unless (member scope '(nil 'tree 'region 'region-start-level 'file))
+             ;; Don't add titles when only showing entries from current buffer
+             (lambda (cand)
+               (let ((marker (car (get-text-property 0 'consult-location cand))))
+                 (buffer-name (marker-buffer marker)))))
+    :narrow (consult-org--narrow)
+    :require-match t
+    :lookup #'consult--lookup-location
+    :history '(:input consult-org--history)
+    :state (consult--jump-preview))))
+
+;;;###autoload
+(defun consult-org-agenda (&optional match)
+  "Jump to an Org agenda heading.
+
+By default, all agenda entries are offered. MATCH is as in
+`org-map-entries' and can used to refine this."
+  (interactive)
+  (unless org-agenda-files
+    (user-error "No agenda files"))
+  (consult-org-heading match 'agenda))
+
+;;;###autoload
+(defun consult-org-clock-in (&optional match scope)
+  "Clock into an Org heading.
+
+MATCH and SCOPE are as in `org-map-entries' and determine which
+entries are offered. By default, offer entries of files with a
+recent clocked item."
+  (interactive)
```

Comment:
Here I wonder about the default behavior. Wouldn't it make sense to use the agenda? Or write two commands `consult-org-clock-in-agenda` and `consult-org-clock-in-recent`?

Reply:
One might often be interested in clocking into agenda entries, but clocking and agenda are two independent features. The first draft of this PR used the agenda, but then I realized there is this alternative way to get a list of clock-relevant files. Do you think it could be confusing / overly smart?

To clock into an agenda entry, one can call `(consult-org-clock-in nil 'agenda)`. Maybe it suffices to let the interested user define this in their config?

Reply:
Where does the original org clocking command gets the clockable files/items? And yes I find it a bit "overly dwim" to scan the history for the clocking relevant files.

Reply:
See https://github.com/minad/consult/pull/276#discussion_r618666135 above.